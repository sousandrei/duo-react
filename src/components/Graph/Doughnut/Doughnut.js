import React from 'react'
import PropTypes from 'prop-types'

import { Doughnut as DoughnutChart } from 'react-chartjs-2'

export const Doughnut = ({
	min = 50,
	max = 50,
	minLabel = 'min',
	maxLabel = 'max',
}) =>
	(
		<div className='doughnut'>
			<div className='doughnut__legend'>
				<div>
					<hr />
					<span>{minLabel}</span>
				</div>
				<div>
					<hr />
					<span>{maxLabel}</span>
				</div>
			</div>
			<DoughnutChart
				data={{
					datasets: [{
						data: [min, max],
						backgroundColor: [
							'#efb524',
							'#36d6cc',
						],
						borderColor: [
							'rgba(0, 0, 0, 0)',
							'rgba(0, 0, 0, 0)'
						]
					}],

					labels: [
						minLabel,
						maxLabel
					]
				}
				}
				options={{ legend: { display: false } }} />
		</div>
	)

Doughnut.propTypes = {
	min: PropTypes.number,
	max: PropTypes.number,
	minLabel: PropTypes.string,
	maxLabel: PropTypes.string,
}


export default Doughnut
