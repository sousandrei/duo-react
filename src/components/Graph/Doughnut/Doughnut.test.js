import React from 'react'
import { shallow } from 'enzyme'

import Doughnut from './Doughnut'

test('should render Doughnut correctly', () => {
	const wrapper = shallow(<Doughnut  />)
	expect(wrapper).toMatchSnapshot()
})
