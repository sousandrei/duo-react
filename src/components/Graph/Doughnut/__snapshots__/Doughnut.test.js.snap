// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Doughnut correctly 1`] = `
<div
  className="doughnut"
>
  <div
    className="doughnut__legend"
  >
    <div>
      <hr />
      <span>
        min
      </span>
    </div>
    <div>
      <hr />
      <span>
        max
      </span>
    </div>
  </div>
  <Doughnut
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": Array [
              "#efb524",
              "#36d6cc",
            ],
            "borderColor": Array [
              "rgba(0, 0, 0, 0)",
              "rgba(0, 0, 0, 0)",
            ],
            "data": Array [
              50,
              50,
            ],
          },
        ],
        "labels": Array [
          "min",
          "max",
        ],
      }
    }
    options={
      Object {
        "legend": Object {
          "display": false,
        },
      }
    }
  />
</div>
`;
